### PR TITLE
Add webserver PodDisruptionBudget

### PR DESCRIPTION
**Pull request by mirobertod**
Add PodDisruptionBudget for the webserver component to helm chart

### CODE REVIEW

**Comment by potiuk** [comment]
LGTM but some tests are faiing. Can you please rebase and see if this is not caused by it ?

**Comment by mirobertod** [comment]
> LGTM but some tests are faiing. Can you please rebase and see if this is not caused by it ?

Sure, rebased.

**Comment by mirobertod** [comment]
> LGTM but some tests are faiing. Can you please rebase and see if this is not caused by it ?

@potiuk tests are green now

**Comment by potiuk** [comment]
I think it would be needed to add helm chart tests for it.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
@jedcunningham ? Do you concur?

File: chart/values.schema.json [hunk 1]
```diff
@@ -2397,6 +2397,30 @@
...
+                            "default": false
+                        },
+                        "config": {
+                            "description": "Disruption budget configuration.",
+                            "type": "object",
+                            "additionalProperties": false,
+                            "properties": {
+                                "maxUnavailable": {
+                                    "description": "Max unavailable pods for webserver.",
+                                    "type": "integer",
```
Comment (jedcunningham) [inline chart/values.schema.json]: This can be an int or a string (e.g. `50%`).

**Review by jedcunningham — CHANGES_REQUESTED** [review]
Overall looks good. What do you think about just exposing `podDisruptionBudget.maxUnavailable`, instead of having the extra `config` layer?

We should also add support for `minAvailable` too.

File: chart/values.schema.json (hunk 1 above)
Comment (mirobertod) [reply chart/values.schema.json]: To keep consistency, I copied the properties from the PDB for the scheduler (https://github.com/apache/airflow/blob/592b46dcad1a2d13e60a49623908e89c161effea/chart/values.schema.json#L1598-L1603)

I can change both if needed

Comment (jedcunningham) [reply chart/values.schema.json]: I see. Yes, let's change both, keep the `config` and `toyaml` it in just like you are, and add `minAvailable` to the schema on both.

Comment (mirobertod) [reply chart/values.schema.json]: Ok, added.

**Review by jedcunningham — APPROVED** [review]

**Comment by mirobertod** [comment]
Tests are failing:

```
chart/tests/test_pdb_pgbouncer.py::PgbouncerPdbTest::test_should_pass_validation_with_just_pdb_enabled_v1:
jsonschema.exceptions.ValidationError: None is valid under each of {'type': ['integer', 'null']}, {'type': ['string', 'null']}

Failed validating 'oneOf' in schema['properties']['spec']['properties']['minAvailable']:
    {'oneOf': [{'type': ['string', 'null']}, {'type': ['integer', 'null']}]}

On instance['spec']['minAvailable']:
    None
```

I don't understand very well the error, I've already added `null` for the list of types in the schema.

@jedcunningham may you help me? thanks

**Comment by jedcunningham** [comment]
I wonder if the null can't land in the actual yaml sent to k8s. That's a k8s validation error, not our schema. Try removing the default `minAvailable: ~` from values.

**Comment by jedcunningham** [comment]
I just noticed that `maxUnavailable` and `minAvailable` are mutually exclusive!

https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/pod-disruption-budget-v1/#PodDisruptionBudgetSpec

Sorry for the detour. I'm thinking we should remove the addition of `minAvailable` for this PR, though still change the type of `maxUnavailable` to int or string.

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @mirobertod 🎉

**Pull request merged**